LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -120,18 +139,20 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                     HandlePreferredGatewayChanges(request, loRaDevice, bundlerResult);
                 }
 
+                #region ADR
                 if (loraPayload.IsAdrAckRequested)
                 {
                     this.logger.LogDebug("ADR ack request received");
                 }
 
-                // ADR should be performed before the deduplication
-                // as we still want to collect the signal info, even if we drop
-                // it in the next step
-                if (loRaADRResult == null && loraPayload.IsDataRateNetworkControlled)
+                // ADR should be performed before the gateway deduplication as we still want to collect the signal info,
+                // even if we drop it in the next step.
+                // ADR is skipped for soft duplicates and will be enabled again in https://github.com/Azure/iotedge-lorawan-starterkit/issues/1017
```

Comment:
I am not even sure we should activate this for duplciates, maybe let's discuss the idea!

Reply:
We can discuss in the linked item if ok for you. We also have an integration test covering exactly this case.